presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java

```diff
@@ -213,8 +210,11 @@ private Type instantiateParametricType(TypeSignature signature)
         if (parametricType == null) {
             throw new IllegalArgumentException("Unknown type " + signature);
         }
+        else if (parametricType instanceof MapParametricType) {
+            return ((MapParametricType) parametricType).createType(functionManager, parameters);
+        }
```

Comment:
What if map is a subtype of some other types. For example `Array<Map<...>>`. `createType` should be recursive.

Reply:
The interesting (or surprising part) is `TypeParameter` already contains `Type`. So it's not recursive. 

https://github.com/prestodb/presto/blob/e6ad497de2d3fba67be46a018878e7648154cd1a/presto-main/src/main/java/com/facebook/presto/type/ArrayParametricType.java#L44-L52